test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -677,28 +842,167 @@ void ProtoConverter::visit(Block const& _x)
+	}
+}
+
+void ProtoConverter::visit(FunctionDefinitionNoReturnVal const& _x)
+{
+	int numInParams = _x.num_input_params() % modInputParams;
+	int numOutParams = 0;
+	createFunctionStub(_x, numInParams, numOutParams, NumFunctionReturns::None);
+}
+
+void ProtoConverter::visit(FunctionDefinitionSingleReturnVal const& _x)
+{
+	int numInParams = _x.num_input_params() % modInputParams;
+	int numOutParams = 1;
+	createFunctionStub(_x, numInParams, numOutParams, NumFunctionReturns::Single);
+}
+
+void ProtoConverter::visit(FunctionDefinitionMultiReturnVal const& _x)
+{
+	int numInParams = _x.num_input_params() % modInputParams;
+	// Synthesize at least 2 return parameters and at most (modOutputParams - 1)
+	int numOutParams = std::max((unsigned int)2, _x.num_output_params() % modOutputParams);
+	createFunctionStub(_x, numInParams, numOutParams, NumFunctionReturns::Multiple);
+}
+
+void ProtoConverter::visit(FunctionDefinition const& _x)
+{
+	visit(_x.fd_zero());
+	visit(_x.fd_one());
+	visit(_x.fd_multi());
+	m_numFunctionSets++;
+}
+
+void ProtoConverter::visit(Program const& _x)
+{
+	/* Program template is as follows
+	 *      Four Globals a_0, a_1, a_2, and a_3 to hold up to four function return values
+	 *
+	 *      Repeated function definitions followed by function calls of the respective function
+	 *          Example: function foo(x_0) -> x_1 {}
+	 *                   a_0 := foo(calldataload(0))
+	 *                   sstore(0, a_0)
+	 */
+	m_output << "{\n";
+	// Create globals at the beginning
+	m_output << "let ";
+	for (int i = 0; i < modOutputParams - 1; i++)
+	{
+		m_output << "a_" << i;
+		if (i < modOutputParams - 2)
+			m_output << ", ";
+		else
+			m_output << "\n";
+	}
+	if (_x.funcs_size() > 0)
+	{
+		// Register function interface. Useful while visiting multi var decl/assignment statements.
+		for (auto const& f: _x.funcs())
+			registerFunction(f);
+
+		for (auto const& f: _x.funcs())
+			visit(f);
+	}
+	yulAssert(_x.funcs_size() == m_numFunctionSets, "Proto fuzzer: Functions not correctly registered.");
 	m_output << "}\n";
 }
 
-string ProtoConverter::functionToString(Function const& _input)
+string ProtoConverter::programToString(Program const& _input)
 {
 	visit(_input);
 	return m_output.str();
 }
 
-string ProtoConverter::protoToYul(const uint8_t* _data, size_t _size)
+void ProtoConverter::registerFunction(FunctionDefinition const& _x)
 {
-	Function message;
-	if (!message.ParsePartialFromArray(_data, _size))
-		return "#error invalid proto\n";
-	return functionToString(message);
-}
+	// No return and single return functions explicitly state the number of values returned
+	registerFunction(_x.fd_zero(), NumFunctionReturns::None);
+	registerFunction(_x.fd_one(), NumFunctionReturns::Single);
+	// A multi return function can have between two and (modOutputParams - 1) parameters
+	int numOutParams = std::max((unsigned int)2, _x.fd_multi().num_output_params() % modOutputParams);
```

Comment:
```suggestion
	int numOutParams = std::max(2u, _x.fd_multi().num_output_params() % modOutputParams);
```

Reply:
or `std::max<unsigned>(2, ...)`

Reply:
done. To keep things consistent, I changed all "number" types in the ProtoConverter class to unsigned.